unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/TeleportPromptHUD/TeleportPromptHUDController.cs

```diff
@@ -13,43 +18,97 @@ public class TeleportPromptHUDController : IHUD
 
     internal TeleportPromptHUDView view { get; private set; }
 
+    private readonly DataStore dataStore;
+    private readonly IMinimapApiBridge minimapApiBridge;
+
     TeleportData teleportData;
+    private CancellationTokenSource cancellationToken = new ();
 
-    public TeleportPromptHUDController()
+    public TeleportPromptHUDController(DataStore dataStore, IMinimapApiBridge minimapApiBridge)
     {
+        this.dataStore = dataStore;
+        this.minimapApiBridge = minimapApiBridge;
+
         view = UnityEngine.Object.Instantiate(Resources.Load<GameObject>("TeleportPromptHUD")).GetComponent<TeleportPromptHUDView>();
         view.name = "_TeleportPromptHUD";
-        view.content.SetActive(false);
+        view.OnCloseEvent += view.SetOutAnimation;
         view.OnTeleportEvent += OnTeleportPressed;
+
+        dataStore.HUDs.gotoPanelVisible.OnChange += ChangeVisibility;
+        dataStore.HUDs.gotoPanelCoordinates.OnChange += SetCoordinates;
     }
 
-    public void SetVisibility(bool visible)
+    private void SetCoordinates(ParcelCoordinates current, ParcelCoordinates previous)
     {
-        if (view.contentAnimator.isVisible && !visible)
+        if (current == previous) return;
+
+        async UniTaskVoid SetCoordinatesAsync(ParcelCoordinates coordinates, CancellationToken cancellationToken)
         {
-            view.contentAnimator.Hide();
+            try
+            {
+                MinimapMetadata.MinimapSceneInfo[] scenes = await minimapApiBridge.GetScenesInformationAroundParcel(new Vector2Int(coordinates.x, coordinates.y),
+                    2,
+                    cancellationToken);
+
+                MinimapMetadata.MinimapSceneInfo sceneInfo = scenes.First(info => info.parcels.Exists(i => i.x == coordinates.x && i.y == coordinates.y));
```

Comment:
IIRC GetSceneInfo only gets already cached info we previously retrieved, with the current architecture this is the only way to obtain the info, it can change once I merge some new functionality with the places API (blocked due to lack of support for some things in the places website and API)